### ISSUE TITLE
Reconstructor configuration not read

### ISSUE THREAD

**Issue by bend-n** [KosmosPrime/plandustry#4]
Reconstructors can be configured to have the unit do a thing.

`bXNjaAF4nBXKPQ6AIAwG0I+fuLjrKTyRcUDo0ASKKdXzq8kbHwJCQJTUCLGxEOZCIytfxl0ATDWdVAf8fjgsqRQ2fmhTyl2G6Z2t69fcz62I/gVp0BSJ`

This schematic causes parsing to fail
```
Could not read schematic: failed to read block data
        Source: invalid dynamic data type (23)
```
See [here](https://github.com/Anuken/Mindustry/blame/3157ff8cea2990011824b77c1570661db5a0543a/core/src/mindustry/io/TypeIO.java)

**Comment by KosmosPrime** [comment]
How did you get this schematic? When I try to command a reconstructor it always issues a move command (which the unit doesn't even follow) and is not serialized to schematics. On the other hand, the game clearly understands the config as it's preserved in the above schematic when built & then copied (but the resulting units don't mine either).

**Comment by bend-n** [comment]
You need to have a unit inside the reconstructor in order to change it, because otherwise mindustry doesnt know what options the unit has.
The units do mine, if they have mining capability (eg poly).

![image](https://github.com/KosmosPrime/plandustry/assets/70787919/088a6a62-d272-4ed2-b624-f9dff5575438)

**Comment by bend-n** [comment]
Ive implemented this on [my fork](https://github.com/bend-n/mindus/commit/5fead62ef30915ab8c65ca30a751a77ab2612c69#diff-9c21bad8a1773a8e6b6b7ec48a9a9cc91ca653ad281a0b771fb02ab9b36ad2f3R311)

**Comment by KosmosPrime** [comment]
These commands are only available when you've got units selected, and it's currently not possible to select units and buildings at the same time.
Monos immediately start mining yes, but that's because there's simply nothing else they can physically do. If you control one and then respawn it'll immediately go back to mining the closest ore.
TL;DR I cannot reproduce the underlying issue (of type 23 data in schematics), which means I cannot test it properly.
(Tested in build 145.1)

**Comment by bend-n** [comment]
No, those are polys, monos go into the reconstructor.

Steps to reproduce:
Place additive from schem
Place unit source for monos, point into additive
Remove unit source, and power the reconstructor

**Comment by KosmosPrime** [comment]
Polys do not start mining immediately if upgraded using the reconstructor blueprint you provided. That's my problem.

I've been trying those steps, with factories running or not, in no case can I instruct any of them to mine resources after being built.

**Comment by bend-n** [comment]
Thats odd; they do for me. But even if they dont start mining, you can still check the config, no?
So even when its not from the schem it doesnt work? Sounds like a anuke bug.

**Comment by KosmosPrime** [comment]
I have, if I place your schematic and then copy the buildings I get a schematic back which has them set. However, nothing I build myself (without schematic hacks) ends up with these configs.
I'm in sandbox currently, I'll sanity check in campaign but that shouldn't be the issue.

Nope, no command mode in campaign either. Do you also get the icons when you select buildings or only when selecting units?

**Comment by KosmosPrime** [comment]
After messing around with a logic processor I got the menu to appear by ctrl-clicking the thing to change the config. Your schematic now also produces units that mine after being reconstruction.
Sorry for making this so difficult.

**Comment by bend-n** [comment]
Did you implement this for plandustry though?
